src/quick-lint-js/diag/diagnostic-types.h

```diff
@@ -1751,6 +1751,14 @@
               question)                                                         \
           MESSAGE(QLJS_TRANSLATABLE("initializer starts here"), equal))         \
                                                                                 \
+  QLJS_DIAG_TYPE(                                                               \
+      diag_optional_parameter_cannot_be_followed_by_required_parameter,         \
+      "E0375", diagnostic_severity::error,                                      \
+      { source_code_span required_parameter; },                                 \
+      MESSAGE(QLJS_TRANSLATABLE("optional parameter cannot be followed by a "   \
+                                "required parameter"),                          \
+              required_parameter))                                              \
```

Comment:
If the diagnostic message talks primarily about the optional parameter ("optional parameter cannot be ..."), then the diagnostic's squiggly underline should be on the optional parameter (or perhaps the `?`). But you put the squiggly underline on the required parameter instead. I think we should underline the optional parameter:
```suggestion
      { source_code_span optional_parameter; },                                 \
      MESSAGE(QLJS_TRANSLATABLE("optional parameter cannot be followed by a "   \
                                "required parameter"),                          \
              optional_parameter))                                              \
```

Minor: However, I think it's a good idea to include a secondary diagnostic message to point out where the required parameter is:
```suggestion
      { source_code_span optional_parameter; source_code_span required_parameter; },                                 \
      MESSAGE(QLJS_TRANSLATABLE("optional parameter cannot be followed by a "   \
                                "required parameter"),                          \
              optional_parameter)                                               \
        MESSAGE(QLJS_TRANSLATABLE("required parameter here"), required_parameter)) \
```